File: src/test/scala/org/apache/flink/pulsar/SchemaData.scala
/**
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.flink.pulsar

import java.sql.Timestamp
import java.util
import java.util.Calendar

import scala.beans.BeanProperty
import scala.collection.JavaConverters._

object SchemaData {

  val booleanSeq = Seq(true, false, true, true, false)
  val bytesSeq = 1.to(5).map(_.toString.getBytes)

  val cal = Calendar.getInstance()
  cal.clear()
  val dateSeq = (1 to 5).map { i =>
    cal.set(2019, 0, i)
    cal.getTime
  }

  cal.clear()
  val timestampSeq = (1 to 5).map { i =>
    cal.set(2019, 0, i, 20, 35, 40)
    new Timestamp(cal.getTimeInMillis)
  }

  val stringSeq = 1.to(5).map(_.toString)
  val int8Seq = 1.to(5).map(_.toByte)
  val doubleSeq = 1.to(5).map(_.toDouble)
  val floatSeq = 1.to(5).map(_.toFloat)
  val int32Seq = 1.to(5)
  val int64Seq = 1.to(5).map(_.toLong)
  val int16Seq = 1.to(5).map(_.toShort)

  case class Foo(@BeanProperty i: Int, @BeanProperty f: Float, @BeanProperty bar: Bar)
  case class Bar(@BeanProperty b: Boolean, @BeanProperty s: String)

  case class F1(@BeanProperty baz: Baz)

  case class Baz(
      @BeanProperty f: Float,
      @BeanProperty d: Double,
      @BeanProperty mp: util.Map[String, Bar],
      @BeanProperty arr: Array[Bar])

  val fooSeq: Seq[Foo] =
    Foo(1, 1.0.toFloat, Bar(true, "a")) :: Foo(2, 2.0.toFloat, Bar(false, "b")) :: Foo(3, 0, null) :: Nil

  val f1Seq: Seq[F1] =
    F1(
      Baz(
        Float.NaN,
        Double.NaN,
        Map("1" -> Bar(true, "1"), "2" -> Bar(false, "2")).asJava,
        Array(Bar(true, "1"), Bar(true, "2")))) ::
    F1(
      Baz(
        Float.NegativeInfinity,
        Double.NegativeInfinity,
        Map("" -> Bar(true, "1")).asJava,
        null)) ::
    F1(Baz(Float.PositiveInfinity, Double.PositiveInfinity, null, null)) ::
    F1(Baz(1.0.toFloat, 2.0, null, null)) :: Nil

  val f1Results = f1Seq.map(f1 =>
    (f1.baz.f, f1.baz.d, if (f1.baz.mp == null) null else f1.baz.mp.asScala, f1.baz.arr))
}

File: src/test/java/org/apache/flink/streaming/connectors/pulsar/SchemaData.java
/**
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.flink.streaming.connectors.pulsar;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
import java.util.Arrays;
import java.util.Calendar;
import java.util.Date;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

public class SchemaData {

    public static final List<Boolean> booleanList = Arrays.asList(true, false, true, true, false);
    public static final List<Integer> int32List = Arrays.asList(1, 2, 3, 4, 5);
    public static final List<byte[]> bytesList = int32List.stream().map(i -> i.toString().getBytes()).collect(Collectors.toList());
    public static final List<Byte> int8List = int32List.stream().map(Integer::byteValue).collect(Collectors.toList());
    public static final List<Short> int16List = int32List.stream().map(Integer::shortValue).collect(Collectors.toList());
    public static final List<Long> int64List = int32List.stream().map(Integer::longValue).collect(Collectors.toList());
    public static final List<Double> doubleList = int32List.stream().map(Integer::doubleValue).collect(Collectors.toList());
    public static final List<Float> floatList = int32List.stream().map(Integer::floatValue).collect(Collectors.toList());
    public static final List<String> stringList = int32List.stream().map(Objects::toString).collect(Collectors.toList());
    public static List<Date> dateList;
    public static List<Timestamp> timestampList;
    public static List<Foo> fooList;

    static {
        Calendar cal = Calendar.getInstance();
        cal.clear();
        dateList = int32List.stream().map(i -> {
            cal.set(2019, 0, i);
            return cal.getTime();
        }).collect(Collectors.toList());

        cal.clear();
        timestampList = int32List.stream().map(i -> {
            cal.set(2019, 0, i, 20, 35, 40);
            return new Timestamp(cal.getTimeInMillis());
        }).collect(Collectors.toList());

        fooList = Arrays.asList(
                new Foo(1, 1.0f, new Bar(true, "a")),
                new Foo(2, 2.0f, new Bar(false, "b")),
                new Foo(3, 0, null),
                new Foo(0, 0, null));
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Foo {
        public int i;
        public float f;
        public Bar bar;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Bar {
        public boolean b;
        public String s;
    }
}
